pkg/deployment/deployment_inspector.go

```diff
@@ -59,13 +83,22 @@ var (
 func (d *Deployment) inspectDeployment(lastInterval util.Interval) util.Interval {
 	log := d.deps.Log
 	start := time.Now()
+
+	timeout, err := d.getReconciliationTimeout()
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to get nodes")
+		return minInspectionInterval // Retry ASAP
+	}
+
+	ctxReconciliation, cancelReconciliation := context.WithTimeout(context.Background(), timeout)
 	defer func() {
+		cancelReconciliation()
```

Comment:
Use 2 defers, make it consistent across PR

Reply:
changed